File: LogicalThinkingPatterns/NumberedRectanglePattern/Numberedrectangle.py
class NumberedRectangle:
    def printPattern(self, n):
        for i in range(0, 2* n - 1):
            pattern = ""
            for j in range(0, 2 * n - 1):
                top = i
                bottom = 2* n - 2 - i
                left = j
                right = 2 * n - 2 - j
                pattern += str(n - min(min(top,bottom), min(left, right)))+" "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

numberedRectangle = NumberedRectangle()
numberedRectangle.run()

File: LogicalThinkingPatterns/NumberedRectanglePattern/NumberedRectangle.ts
class NumberedRectangle {
  printPattern(n: number): void {
    for (let i = 0; i < 2 * n - 1; i++) {
      let pattern: string = "";
      for (let j = 0; j < 2 * n - 1; j++) {
        let top: number = i;
        let left: number = j;
        let bottom: number = 2 * n - 2 - i;
        let right: number = 2 * n - 2 - j;
        pattern +=
          n - Math.min(Math.min(top, bottom), Math.min(left, right)) + " ";
      }
      console.log(pattern);
    }
  }
}

const numberedRectangle = new NumberedRectangle();
numberedRectangle.printPattern(4);
